third_party/move/documentation/book/src/loops.md

```diff
@@ -114,6 +114,45 @@ fun pick(
 }
 ```
 
+## The `for` expression
+
+The `for` expression iterates over a range defined using a `lower_bound` (inclusive) and an `upper_bound` (non-inclusive) of type integer, executing its body for each element. Unlike `while`, `for` is designed for scenarios where the number of iterations is determined by a specific range.
+
+Here is an example of a `for` loop that computes the sum of the elements in a range from `0` to `n`:
+
+```move
+fun sum(n: u64): u64 {
+    let sum = 0;
+    for (i in 0..(n + 1)) {
```

Comment:
I would suggest using the simplest range expressions while first introducing `for`, so please consider changing to `0..n` (and adjust surrounding explanatory text correspondingly).

Reply:
Thanks. I have made the change in this PR: https://github.com/aptos-labs/developer-docs/pull/46